### PR TITLE
Track generator changes for current collector only

### PR DESCRIPTION
**Pull request by jgraff2**
Previously the changes were tracked based on Generator.collectors. This uses Generator.currentCollector instead.

Because currentCollector can't be updated directly through the API the calls to trackGeneratorChanges are moved to db hooks, and the tests are updated to not go through the api.

Actually setting the currentCollector field will be done in the other story I have this sprint.

### CODE REVIEW

File: db/model/generator.js [hunk 1]
```diff
@@ -191,16 +192,36 @@ module.exports = function generator(seq, dataTypes) {
...
+              return inst.addWriter(inst.createdBy);
+            }
+          }),
+        ]);
 
         return inst;
       }, // afterCreate
+
+      afterUpdate(inst) {
+        let oldCollector = inst.previous('currentCollector');
```
Comment (iamigo) [inline db/model/generator.js]: check `inst.changed()` for whether currentCollector or isActive were changed before doing anything here... may be able to short circuit and exit without doing anything

Comment (jgraff2) [reply db/model/generator.js]: If currentCollector wasn't changed we still need to track the changes based on whether isActive changed. (line 216 & 218)
And if isActive wasn't changed we still need to check if it's false, in which case we don't track anything. (line 220)

File: api/v1/helpers/verbs/heartbeatUtils.js [hunk 2]
```diff
@@ -102,28 +102,26 @@ function getKey(collectorName, changeType) {
...
+  }
 
-      // determine the change type
-      let change;
-      const inOld = oldCollectors.includes(collectorName);
-      const inNew = newCollectors.includes(collectorName);
-      if (inOld && !inNew) change = DELETED;
-      if (!inOld && inNew) change = ADDED;
-      if (inOld && inNew) change = UPDATED;
+  function trackChangesForCollector(collectorName, change, generator) {
```
Comment (pallavi2209) [inline api/v1/helpers/verbs/heartbeatUtils.js]: Defining trackChangesForCollector function inside trackGeneratorChanges function is resulting in multiple eslint errors on my sublime - I would like to know if there is an advantage in doing so.

Comment (jgraff2) [reply api/v1/helpers/verbs/heartbeatUtils.js]: Personally I much prefer using internal functions in a case like this where it isn't used anywhere else and makes no sense outside the context of the calling function. It keeps it self-contained and makes it easier to understand.

Comment (jgraff2) [reply api/v1/helpers/verbs/heartbeatUtils.js]: You already approved so I will go ahead and merge if there are no objections.

Comment (pallavi2209) [reply api/v1/helpers/verbs/heartbeatUtils.js]: Please do. For some reason, squash and merge is not enabled for me right now.

**Pull request merged**